fdbclient/GlobalConfig.actor.cpp

```diff
@@ -155,91 +155,26 @@ void GlobalConfig::erase(KeyRangeRef range) {
 
 // Older FDB versions used different keys for client profiling data. This
 // function performs a one-time migration of data in these keys to the new
-// global configuration key space.
+// global configuration key space (proxied through the GrvProxies).
 ACTOR Future<Void> GlobalConfig::migrate(GlobalConfig* self) {
-	state Key migratedKey("\xff\x02/fdbClientInfo/migrated/"_sr);
-	state Reference<ReadYourWritesTransaction> tr;
-	try {
-		state Backoff backoff;
-		loop {
-			tr = makeReference<ReadYourWritesTransaction>(Database(Reference<DatabaseContext>::addRef(self->cx)));
-			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-
-			try {
-				state Optional<Value> migrated = wait(tr->get(migratedKey));
-				if (migrated.present()) {
-					// Already performed migration.
-					return Void();
-				}
-
-				state Optional<Value> sampleRate =
-				    wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_sample_rate/"_sr)));
-				state Optional<Value> sizeLimit =
-				    wait(tr->get(Key("\xff\x02/fdbClientInfo/client_txn_size_limit/"_sr)));
-
-				tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
-				// The value doesn't matter too much, as long as the key is set.
-				tr->set(migratedKey.contents(), "1"_sr);
-				if (sampleRate.present()) {
-					const double sampleRateDbl =
-					    BinaryReader::fromStringRef<double>(sampleRate.get().contents(), Unversioned());
-					Tuple rate = Tuple().appendDouble(sampleRateDbl);
-					tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSampleRate), rate.pack());
-				}
-				if (sizeLimit.present()) {
-					const int64_t sizeLimitInt =
-					    BinaryReader::fromStringRef<int64_t>(sizeLimit.get().contents(), Unversioned());
-					Tuple size = Tuple().append(sizeLimitInt);
-					tr->set(GlobalConfig::prefixedKey(fdbClientInfoTxnSizeLimit), size.pack());
-				}
-
-				wait(tr->commit());
-				break;
-			} catch (Error& e) {
-				// If multiple fdbserver processes are started at once, they will all
-				// attempt this migration at the same time, sometimes resulting in
-				// aborts due to conflicts. Purposefully avoid retrying, making this
-				// migration best-effort.
-				TraceEvent(SevInfo, "GlobalConfig_RetryableMigrationError").errorUnsuppressed(e).suppressFor(1.0);
-				wait(tr->onError(e));
-				tr.clear();
-				// tr is cleared, so it won't backoff properly. Use custom backoff logic here.
-				wait(backoff.onError());
-			}
-		}
-	} catch (Error& e) {
-		// Catch non-retryable errors (and do nothing).
-		TraceEvent(SevWarnAlways, "GlobalConfig_MigrationError").error(e);
-	}
+	wait(basicLoadBalance(self->cx->getGrvProxies(UseProvisionalProxies::False),
+	                      &GrvProxyInterface::migrateGlobalConfig,
+	                      GlobalConfigMigrateRequest{}));
```

Comment:
Since this only needs to be done once, could we have the proxy attempt it when it starts up the `globalConfigRequestServer` rather than having the clients send a request to do the migration?

Reply:
Yeah I think this makes a lot of sense!